### PR TITLE
Update NaN.java

### PR DESCRIPTION
**Pull request by nimo23**
Add method valueOf()

Changes proposed in this pull request:
- added method valueOf() (needed for Builder, be consistent with DoubleNum, DecimalNum)

With valueOf() it is possible to use the Builder, for example:

`BaseBar.builder(NaN::valueOf, NaN.class).timePeriod(..).openPrice(NaN).build();`

- [ ] added an entry with related ticket number(s) to the unreleased section of `CHANGES.md`

### CODE REVIEW

**Comment by nimo23** [comment]
@team172011 something is wrong with the license header..and please look at other classes, for example: DecimalNum.java, there is a duplicate license header..does the formatter plugin adds an additional the license header?

**Comment by nimo23** [comment]
@team172011 On another questions.: In `DoubleNum` there is the following apiNote:

> the delegate should never become a NaN value.

Can you explain the reason for this a little more? 

There is a `Double.NaN` but it has nothing to do with our Num's `NaN`. And why is this apiNote missing in DecimalNum (which could also have BigDecimal.NaN)?

File: ta4j-core/src/main/java/org/ta4j/core/num/NaN.java [hunk 1]
```diff
@@ -50,6 +50,16 @@
...
     private NaN() {
     }
+    
+    /**
+     * Returns a {@code Num} version of the given {@code Num}.
+     *
+     * @param val the number
+     * @return the {@code Num}
+     */
+    public static NaN valueOf(NaN val) {
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/num/NaN.java]: I would change signature to `valueOf(Number val) -> NaN`

Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/num/NaN.java]: Do you mean this:

```
    public static Num valueOf(Number val) {
        return NaN;
    }
```

or that:

```
 public static Num valueOf(Num val) {
        return NaN;
    }
```
?

Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/num/NaN.java]: Either we can `valueOf(Num num)` or `valueOf(NaN val)` to be usable for the builder

`BaseBar.builder(org.ta4j.core.num.NaN::valueOf, NaN.class).openPrice(NaN).build();`

I changed the code to use `valueOf(Num num)`.

Comment (team172011) [reply ta4j-core/src/main/java/org/ta4j/core/num/NaN.java]: I mean 
```
    public static Num valueOf(Number val) {
        return NaN;
    }
```
to be more general

**Comment by team172011** [comment]
>the delegate should never become a NaN value.

It is a very old api note when we were using only one wrapper class. I think it should ensure that we are using our NaN and not working with Double.NaN or BigDecimal.NaN under the hood and getting unexpected behaviour.

**Pull request merged**